src/main/kotlin/org/jetbrains/kotlinx/jupyter/apiImpl.kt

```diff
@@ -132,6 +136,23 @@ class NotebookImpl(
     override val jreInfo: JREInfoProvider
         get() = JavaRuntime
 
+    fun updateVarsState(varsMap: Map<String, VariableState>) {
+        variablesMap += varsMap
+    }
+
+    fun varsAsHtmlTable(): String {
+        return generateHTMLVarsReport(variablesMap)
+    }
+
+    fun varsAsString(): String {
```

Comment:
Maybe `variablesReport`? It's not just stringified variable values, it's a report, with a header and variable names

Reply:
Fair's fair